compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -46,6 +46,7 @@ class Compiler {
     List(new sjs.PrepJSInterop) ::  // Additional checks and transformations for Scala.js (Scala.js only)
     List(new sbt.ExtractAPI) ::     // Sends a representation of the API of classes to sbt via callbacks
     List(new SetRootTree) ::        // Set the `rootTreeOrProvider` on class symbols
+    List(new CoverageTransformMacro) :: // Perform instrumentation for coverage transform (if -coverage is present)
```

Comment:
It would be petter to place this after the Pickler phase. This way we can take some existing tasty and compile with or without coverage.

Maybe it should be at the end of the picklerPhases. This way generated code from inclining and macros can also be part of coverage.

Reply:
huh, I originally put it there since in the original report in [here](http://guillaume.martres.me/code_coverage.pdf) it said:

> The first question that arises when we implement a new compiler phase is to choose where to add it. In
the context of code coverage, the earlier is the better in order to not lose any information that we need
to give a useful feedback to the user. I choose to add it at the end of the front-end phase, just before the
phases dealing with TASTY tree pickling and unpickling. It is important to do it before phases that
perform some optimization like dead code elimination. For example, the transformation FirstTransform
is already removing some unreachable code from If tree with constant condition, this is typically some
code that we want to instrument before being removed by the different optimization of the compiler as
we want to show it as uncovered even if it is removed. 

So I sort of just went with that since it seemed to make sense.